src/internal/operators/delay.ts

```diff
@@ -92,6 +92,7 @@ class DelaySubscriber<T> extends Subscriber<T> {
       const delay = Math.max(0, queue[0].time - scheduler.now());
       this.schedule(state, delay);
     } else {
+      source.remove(this);
```

Comment:
I've changed my mind with this. I think `this.unsubscribe()` instead of `source.remove(this)` would be better. The effected behaviour will be identical, but the former is more idiomatic.